src/averageAges.js

```diff
@@ -55,7 +90,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
```

Comment:
Try to refactor this function so it is more readable, avoid mutating the arrays.
You can first filter people to find children and then count age differences.

Reply:
Better option is to count sum ages differences in filter, because I have access to children and their mothers.